src/storage-controller/src/persist_handles.rs

```diff
@@ -594,8 +594,7 @@ impl<T: Timestamp + Lattice + Codec64 + TimestampManipulation> TxnsTableWorker<T
                 // - Reboot.
                 // - Try and read s at 10.
                 if !self.lazy_persist_txn_tables {
-                    self.tidy
-                        .merge(self.txns.apply_eager_le(&register_ts).await);
+                    self.tidy.merge(tidy);
```

Comment:
I think we'd want to merge the tidy in either case, no?

Reply:
Yes, fixed.